feed2discord.py

```diff
@@ -460,427 +457,60 @@ async def actually_send_message(channel, message, delay, FEED, feed):
-                    http_headers["ETag"] = etag
-                else:
-                    logger.info(feed + ":no stored ETag")
-
-            # Set up httpclient
-            httpclient = aiohttp.ClientSession()
-
-            logger.info(feed + ":sending http request for " + feed_url)
-            # Send actual request.  await can yield control to another
-            # instance.
-            http_response = await httpclient.get(feed_url, headers=http_headers)
-
-            logger.info(http_response)
-
-            # First check that we didn't get a "None" response, since that's
-            # some sort of internal error thing:
-            if http_response.status is None:
-                logger.error(feed + ":HTTP response code is NONE")
-                raise HTTPError()
-            # Some feeds are smart enough to use that if-modified-since or
-            # etag info, which gives us a 304 status.  If that happens,
-            # assume no new items, fall through rest of this and try again
-            # later.
-            elif http_response.status == 304:
-                logger.info(feed + ":data is old; moving on")
-                http_response.close()
-                raise HTTPNotModified()
-            # If we get anything but a 200, that's a problem and we don't
-            # have good data, so give up and try later.
-            # Mostly handled different than 304/not-modified to make logging
-            # clearer.
-            elif http_response.status != 200:
-                logger.info(feed + ":HTTP error not 200")
-                http_response.close()
-                raise HTTPNotModified()
-            else:
-                logger.info(feed + ":HTTP success")
-
-            # pull data out of the http response
-            logger.info(feed + ":reading http response")
-            http_data = await http_response.read()
-
-            # Apparently we need to sleep before closing an SSL connection?
-            # https://docs.aiohttp.org/en/stable/client_advanced.html#graceful-shutdown
-            await asyncio.sleep(5)
-            await httpclient.close()
-
-            # parse the data from the http response with feedparser
-            logger.info(feed + ":parsing http data")
-            feed_data = feedparser.parse(http_data)
-            logger.info(feed + ":done fetching")
-
-            # If we got an ETAG back in headers, store that, so we can
-            # include on next fetch
-            if "ETAG" in http_response.headers:
-                etag = http_response.headers["ETAG"]
-                logger.info(feed + ":saving etag: " + etag)
-                conn.execute(
-                    "UPDATE feed_info SET etag=? where feed=? or url=?",
-                    [etag, feed, feed_url],
-                )
-                conn.commit()
-                logger.info(feed + ":etag saved")
-            else:
-                logger.info(feed + ":no etag")
+class CheckFeed():
```

Comment:
Hmmm...  thinking maybe this should be "WatchFeed" instead of "CheckFeed"? It's an infinite loop of feed checks...